examples/multiclass/plot_multiclass_overview.py

```diff
@@ -0,0 +1,196 @@
+# tree classifier is working quite well. One-vs-one and the error-correcting
+# output code strategies are working even better. However, the
+# one-vs-rest strategy is not working as well as the other strategies.
+#
+# Indeed, these results reproduce something reported in the literature
+# as in [2]_. However, the story is not as simple as it seems.
+#
+# The importance of hyperparameters search
+# ----------------------------------------
+#
+# It was later shown in [3]_ that the multiclass strategies would show similar
+# scores if the hyperparameters of the base classifiers are first optimized.
+#
+# Here we try to reproduce such result by at least optimizing the depth of the
+# base decision tree.
+from sklearn.model_selection import GridSearchCV
+
+param_grid = {"max_depth": [3, 5, 8]}
+tree_optimized = GridSearchCV(tree, param_grid=param_grid, cv=3)
+ovo_tree = OneVsOneClassifier(tree_optimized)
+ovr_tree = OneVsRestClassifier(tree_optimized)
+ecoc = OutputCodeClassifier(tree_optimized, code_size=2)
+
+cv_results_tree = cross_validate(tree_optimized, X, y, cv=cv, n_jobs=2)
+cv_results_ovo = cross_validate(ovo_tree, X, y, cv=cv, n_jobs=2)
+cv_results_ovr = cross_validate(ovr_tree, X, y, cv=cv, n_jobs=2)
+cv_results_ecoc = cross_validate(ecoc, X, y, cv=cv, n_jobs=2)
+
+scores = pd.DataFrame(
+    {
+        "DecisionTreeClassifier": cv_results_tree["test_score"],
+        "OneVsOneClassifier": cv_results_ovo["test_score"],
+        "OneVsRestClassifier": cv_results_ovr["test_score"],
+        "OutputCodeClassifier": cv_results_ecoc["test_score"],
+    }
+)
+ax = scores.plot.kde(legend=True)
+ax.set_xlabel("Accuracy score")
+ax.set_xlim([0, 0.7])
+_ = ax.set_title(
+    "Density of the accuracy scores for the different multiclass strategies"
+)
+
+plt.show()
+
+# %%
+# We can see that once the hyperparameters are optimized, all multiclass
+# strategies have similar performance as discussed in [3]_.
+#
+# Conclusion
+# ----------
+#
+# We can get some intuition behind those results.
+#
+# First, the reason for which one-vs-one and error-correcting output code are
+# outperforming the tree when the hyperparameters are not optimized relies on
+# fact that they ensemble a larger number of classifiers. The ensembling
+# improves the generalization performance. This is a bit similar why a bagging
+# classifier generally performs better than a single decision tree if no care
+# is taken to optimize the hyperparameters.
+#
+# Then, we see the importance of optimizing the hyperparameters. Indeed, it
+# should be regularly explored when developing predictive models even if
+# techniques such as ensembling help at reducing this impact.
+#
+# Finally, it is important to recall that the estimators in scikit-learn
+# are developed with a specific strategy to handle multiclass classification
+# out of the box. It means that there is little interest in using the
+# different strategies. However, it also shows that even if a strategy has been
```

Comment:
> It means that there is little interest in using the different strategies.

This statement demotivates the whole example. I think the use case for the `multiclass` module is to extend third party estimators that only support binary classification.

Reply:
I see I can make a more gentle transition mentioning that the strategies is probably not where one should put an effort but rather use one of them and advise for parameter tuning.